compiler/rustc_middle/src/ty/print/pretty.rs

```diff
@@ -637,7 +637,9 @@ pub trait PrettyPrinter<'tcx>:
                 p!(print_def_path(def_id, &[]));
             }
             ty::Projection(ref data) => {
-                if self.tcx().def_kind(data.item_def_id) == DefKind::ImplTraitPlaceholder {
+                if self.tcx().def_kind(data.item_def_id) == DefKind::ImplTraitPlaceholder
+                    && !(self.tcx().sess.verbose() || NO_QUERIES.with(|q| q.get()))
```

Comment:
As `def_kind` is a query, should the test be reversed to short-circuit it?
Otherwise r=me.